atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/CoordinationStore.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.coordination;
+
+import java.util.Optional;
+
+import com.palantir.atlasdb.keyvalue.impl.CheckAndSetResult;
+
+import okio.ByteString;
+
+/**
+ * A {@link CoordinationStore} stores data that a {@link CoordinationService} may use.
+ */
+public interface CoordinationStore {
```

Comment:
Alternately, we could hide the sequence number from users of the store - putting a value does the equivalent of `putValue` followed by `checkAndSetCoordinationValue`. As it is, it's not immediately clear what the sequence number represents, and it doesn't look like it's meaningful outside of being used in `getValue`. If `getCoordinationValue` returned a `ByteStreamAndBound`, would that remove the need? One downside would be that the client can't cache values of known sequence numbers, but we could put that logic in the store (alongside disallowing updates to existing sequence numbers, which we can easily do internally).

Reply:
Discussed offline - schema changed to `ValueAndBound<T>`. The idea makes sense and contains, though I chose to use `T` rather than `ByteStreamAndBound` as having users provide transforms from `ByteStream -> ByteStream` seemed a bit excessive.